src/sentry/models/groupsubscription.py

```diff
@@ -121,6 +131,26 @@ def bulk_subscribe(
                 if user_id not in existing_subscriptions
             ]
 
+            existing_team_subscriptions = set(
+                GroupSubscription.objects.filter(
+                    team__in=teams, group=group, project=group.project
+                ).values_list("team", flat=True)
+            )
+
+            subscriptions.extend(
+                [
+                    GroupSubscription(
+                        team=team,
+                        group=group,
+                        project=group.project,
+                        is_active=True,
+                        reason=reason,
+                    )
+                    for team in teams
```

Comment:
you could use `teams.difference(existing_team_subscriptions)` instead of a list comprehension to filter since they're sets https://python-reference.readthedocs.io/en/latest/docs/sets/difference.html

Reply:
The issue is that `existing_team_subscriptions` only returns a set of the team ids, rather than a set of teams. So the difference between `teams` and `existing_team_subscriptions` is equivalent to `teams` as nothing gets filtered out. Additionally, the same logic is used to deal with users.